### PR TITLE
Address cmake compilation

### PR DESCRIPTION
**Pull request by thiagoftsm**
##### Summary
After @stelfrag  reports, this PR is addressing some issues while we are compiling with commands:

```sh
$ cmake -S . -B mytest -DCMAKE_INSTALL_PREFIX=/home/install_path -DCMAKE_BUILD_TYPE=Debug -DENABLE_PLUGIN_LOGS_MANAGEMENT=Off -DENABLE_LOGS_MANAGEMENT_TESTS=Off -DENABLE_BUNDLED_JSONC=Off -DENABLE_BUNDLED_YAML=off -DENABLE_PLUGIN_XENSTAT=off -DENABLE_PLUGIN_EBPF=On -DENABLE_ACLK=off -DENABLE_EXPORTER_PROMETHEUS_REMOTE_WRITE=off
$ cmake --build mytest --target install
```
and also normal user.

##### Test Plan

1. Compile locally as normal user. You should not have issues.

##### Additional Information

<details> <summary>For users: How does this change affect me?</summary>
  
</details>

### CODE REVIEW

**Comment by stelfrag** [comment]
@thiagoftsm this seems to work ok now

File: packaging/cmake/Modules/NetdataDetectSystemd.cmake [hunk 1]
```diff
@@ -38,5 +38,7 @@ macro(detect_systemd)
        if(HAVE_SD_BUS_DEFAULT_SYSTEM AND HAVE_SD_BUS_CALL_METHOD AND HAVE_SD_BUS_MESSAGE_ENTER_CONTAINER AND HAVE_SD_BUS_MESSAGE_READ AND HAVE_SD_BUS_MESSAGE_EXIT_CONTAINER)
            set(ENABLE_SYSTEMD_DBUS YES)
        endif()
+   else()
+       set(ENABLE_PLUGIN_SYSTEMD_JOURNAL NO)
```
[the remark this answers is not in the document]
Comment (thiagoftsm) [reply packaging/cmake/Modules/NetdataDetectSystemd.cmake]: Done.

**Comment by thiagoftsm** [comment]
Thank you @Ferroin !

Guys, the PR is ready for review again, please, take a look when you have a time.

**Review by Ferroin — APPROVED** [review]
CMake changes LGTM, I have not tested the C changes.

**Pull request merged**